rock-n-rollup/src/core/runtime.rs

```diff
@@ -104,132 +49,127 @@ pub trait Runtime: 'static {
 }
 
 #[derive(Default)]
-pub struct KernelRuntime {}
+pub struct KernelRuntime<R>
+where
+    R: tezos_smart_rollup_host::runtime::Runtime + 'static,
+{
+    host: R,
+}
 
-impl Runtime for KernelRuntime {
-    fn write_debug(&mut self, msg: &str) {
-        unsafe {
-            write_debug(msg.as_ptr(), msg.len());
-        }
+impl<R> KernelRuntime<R>
+where
+    R: tezos_smart_rollup_host::runtime::Runtime,
+{
+    pub fn new(host: R) -> Self {
+        Self { host }
     }
+}
 
-    fn next_input(&mut self) -> Option<RawInput> {
-        let mut payload = Vec::with_capacity(MAX_MESSAGE_SIZE as usize);
-
-        // Placeholder values
-        let mut message_info = ReadInputMessageInfo { level: 0, id: 0 };
+impl<R> Runtime for KernelRuntime<R>
+where
+    R: tezos_smart_rollup_host::runtime::Runtime,
+{
+    fn write_debug(&mut self, msg: &str) {
+        self.host.write_debug(msg)
+    }
 
-        let size = unsafe { read_input(&mut message_info, payload.as_mut_ptr(), MAX_MESSAGE_SIZE) };
+    fn store_delete(&mut self, path: &str) -> Result<(), ()> {
+        let path = OwnedPath::try_from(path.to_string()).map_err(|_| ())?;
 
-        if size == 0 {
-            None
-        } else {
-            unsafe { payload.set_len(size as usize) };
-            Some(RawInput {
-                level: message_info.level as u32,
-                id: message_info.id as u32,
-                payload,
-            })
+        let res = self.host.store_delete(&path);
+        match res {
+            Ok(_) => Ok(()),
+            Err(_) => Err(print!("Error store_delete")),
```

Comment:
You should not use print!, if you want to print a message you can directly use `write_debug`.

(same in other places in the code)

Reply:
I replace `print!` with `write_debug` as this [commit](f62e693698d8b622c3351f686d7de6cd33b20721)